Automattic/vip-go-mu-plugins
# Fix incorrect limit return for wpcom_vip_top_posts_array

## Description

Due to how WordPress.com caches these requests, `wpcom_vip_top_posts_array` can return incomplete data.  WordPress.com will cache these by number of days and not by limit.

Example:
```
wpcom_vip_top_posts_array( 5, 10 );   // 10 posts returned
wpcom_vip_top_posts_array( 5, 100 );  // 10 posts returned
```

In this change, we set the `limit` passed to WordPress.com to always be 100, so we'll always get the maximum amount of posts available.  This function will then only return the amount based on the `limit` that it was passed.

## Changelog Description

### Bug Fix: Fixed an issue where wpcom_vip_top_posts_array() was not always returning the correct amount of posts

Previously, it was possible for our requests to return cached results with the wrong number of posts. This has been rectified.

## Checklist

Please make sure the items below have been covered before requesting a review:

- n/a This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. Check out PR.
2. Sandbox a VIP Go test site, make sure it has visits!  
3. If you need to spoof visits - open up an Incognito/Private window and just reload a bunch.  Or run an Apache Bench command. 
2. Open up WP_CLI and use `wp shell`
3. Run `wpcom_vip_top_posts_array( 5, 10 );`  // Did you get 10 posts?
4. Now run `wpcom_vip_top_posts_array( 5, 20 );` // Did you get 20 posts?
